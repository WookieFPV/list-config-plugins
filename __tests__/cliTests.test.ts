import { describe, expect, it } from "bun:test";
import { $ } from "zx";

/**
 * These E2E Test use a real package json + app.config.js and a node_modules folder where everything is removed expect the config plugins files (this way CI is much faster because we can add those files to git instead of installing packages)
 */
describe("CLI Tests", () => {
    it("should show the correct config plugins", async () => {
        const { stdout: node } = await $`node -v`;
        console.log(`node ${node.trim()}`);

        const { stdout: lcp } = await $`npx -y list-config-plugins@latest -v`;
        console.log(`list-config-plugins: ${lcp.trim()}`);

        const { stdout } = await $`npx -y list-config-plugins@latest`;
        expect(stdout).toMatchSnapshot();
    });
});
