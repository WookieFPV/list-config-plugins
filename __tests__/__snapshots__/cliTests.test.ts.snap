// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Tests should show the correct config plugins 1`] = `
"Config Plugin Overview:

Used plugins:
游릴  @bitdrift/react-native 
游릴  expo-build-properties 
游릴  expo-font 
游릴  expo-localization 
游릴  expo-notifications 
游릴  expo-screen-orientation 
游릴  expo-splash-screen 
游릴  expo-video 
游릴  react-native-compressor 
游릴  react-native-edge-to-edge 

Bundled with expo:
游닍  expo-camera 
游닍  expo-dev-client 
游닍  expo-file-system 
游닍  expo-image-picker 
游닍  expo-media-library 
游닍  expo-task-manager 

Unused plugins:
游린  @sentry/react-native 
游린  expo-system-ui 
游린  expo-updates 
"
`;
